homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -1092,6 +1082,58 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
         self._preset_mode = preset_mode
         self.async_write_ha_state()
 
+    async def async_set_percentage(self, percentage: int) -> None:
+        """Set the percentage of the fan."""
+        if percentage == 0:
+            self._percentage = 0
+            await self.async_turn_off()
+            return
+
+        await self._try_command(
```

Comment:
Should we check the result before writing a new state?

Reply:
I could, however, I reverted this code path to the original implementation before the regression. I personally cannot test it. So I could implement it but not test it. Would you like to implement it anyway? I see the risk of breaking stuff low. But I assumed that I don't break stuff with my initial PR too :D

Reply:
Since we're asking for someone to test the PR before merging here I think we can include the change. Not including it looks like a bug.

Reply:
Ok, I'll include it tonight in this PR.

Reply:
Added the change to check for setting percentage. If it was not successful the internal state will not be updated with the new percentage.